### ISSUE TITLE
A problem running teochat_demo.py locally

### ISSUE THREAD

**Issue by BrooksXiaoxi** [ermongroup/TEOChat#3]
1. I cloned the repository from the readme content and downloaded the model weights in hugging face.
2. I then modified these places to make the hugging face work in an offline environment.
![image](https://github.com/user-attachments/assets/4b42651b-08c3-4cce-9375-9ff6568d9c74)
![image](https://github.com/user-attachments/assets/1c61feff-c0e5-45dd-87dc-7074721690c1)
3. I got this error...Can someone tell me how to fix it?
![image](https://github.com/user-attachments/assets/5afe3772-52e6-4b66-93ad-25dc525339be)

**Comment by jirvin16** [comment]
Is that the full traceback?

Also, the changes to `llava_arch.py` should not be necessary. What error were you getting before making that change?

**Comment by BrooksXiaoxi** [comment]
Yes, if I don't modify llava_arch.py will get hugging face error
my code：
![image](https://github.com/user-attachments/assets/9b6371c9-04a4-4cf6-881c-268bd0f179d9)
![image](https://github.com/user-attachments/assets/e33a5a47-a690-448e-84ce-5b232d6b2dd9)
error is：
![image](https://github.com/user-attachments/assets/40542ae5-0543-4d0e-9c28-1710eb7a59e4)
![image](https://github.com/user-attachments/assets/36a7ae2c-1f96-4a1a-9d51-b457b8ae1cf8)
![image](https://github.com/user-attachments/assets/3cd9169e-900e-40a3-9e34-15914e2aee56)
Sometimes I had to manually download all the code base files from hugging face to my local location, which resolved the error most of the time, but this time it didn't work well. Is it caused by other errors?

**Comment by jirvin16** [comment]
It looks like there are potentially several issues here:
1. CUDA setup error. Are you running this on GPU? If not, you need to specify `--device cpu`. 
2. Network connection error when trying to download the models on huggingface. Is the device running the demo connected to the internet?

**Comment by BrooksXiaoxi** [comment]
Thanks for your reply! 
Certainly, I have reinstalled `bitsandbytes==0.41.0` and executed `python -m bitsandbytes` to verify that `bitsandbytes` is installed correctly. Now, CUDA is functioning properly. 
When loading the model weights, it is reported that some weights are not used. Is this a normal occurrence? Thanks. 
![image](https://github.com/user-attachments/assets/4a7188ff-033a-4f75-8717-2becf0ada8f4)
![image](https://github.com/user-attachments/assets/cc7ca99b-08b9-4863-aafa-e3f5ca555238)
![image](https://github.com/user-attachments/assets/a134a981-6bec-4d2a-a2af-85ae48888b80)
I have tested it on a locally deployed web page and the model is responding well. What mistakes could have caused this?
